packages/core/core/src/migrations/draft-publish.ts

```diff
@@ -0,0 +1,108 @@
+import { contentTypes as contentTypesUtils } from '@strapi/utils';
+import { Schema } from '@strapi/types';
+
+interface Input {
+  oldContentTypes: Record<string, Schema.ContentType>;
+  contentTypes: Record<string, Schema.ContentType>;
+}
+
+/**
+ * Enable draft and publish for content types.
+ *
+ * Draft and publish disabled content types will have their entries published,
+ * this migration clones those entries as drafts.
+ *
+ * TODO: Clone components, dynamic zones and relations
+ */
+const enableDraftAndPublish = async ({ oldContentTypes, contentTypes }: Input) => {
+  if (!oldContentTypes) {
+    return;
+  }
+
+  // run the after content types migrations
+
+  for (const uid in contentTypes) {
+    if (!oldContentTypes[uid]) {
+      continue;
+    }
+
+    const oldContentType = oldContentTypes[uid];
+    const contentType = contentTypes[uid];
+
+    // if d&p was enabled set publishedAt to eq createdAt
+    if (
+      !contentTypesUtils.hasDraftAndPublish(oldContentType) &&
+      contentTypesUtils.hasDraftAndPublish(contentType)
+    ) {
+      const metadata = strapi.db.metadata.get(uid);
+
+      // Extract all scalar attributes to use in the insert query
+      const attributes = Object.values(metadata.attributes).reduce((acc, attribute: any) => {
+        if (['id'].includes(attribute.columnName)) {
+          return acc;
+        }
+
+        if (contentTypesUtils.isScalarAttribute(attribute)) {
+          acc.push(attribute.columnName);
+        }
+
+        return acc;
+      }, [] as string[]);
+
+      /**
+       * INSERT INTO tableName (columnName1, columnName2, columnName3, ...)
+       * SELECT columnName1, columnName2, columnName3, ...
+       * FROM tableName
+       */
+      const qb = strapi.db?.getConnection();
+      await qb
+        // INSERT INTO tableName (columnName1, columnName2, columnName3, ...)
+        .into(qb.raw(`${metadata.tableName} (${attributes.join(', ')})`))
+        .insert((subQb: typeof qb) => {
+          // SELECT columnName1, columnName2, columnName3, ...
+          subQb
+            .select(
+              ...attributes.map((att) => {
+                // Override 'publishedAt' and 'updatedAt' attributes
+                if (att === 'published_at') {
+                  return qb.raw('NULL as published_at');
+                }
+
+                if (att === 'updated_at') {
+                  return qb.raw(`'${new Date().getTime()}' as updated_at`);
```

Comment:
I'm afraid this will not work for all DBs datetime are note always stored as timestamps 🤔

Reply:
maybe use a binding instead of knex does the right transformation for us 

qb.raw(`?, as updated_at`, new Date())

Reply:
will try!

Reply:
new Date does not work, but I saw a a [`knex.fn.now()`](https://knexjs.org/guide/utility.html#now) utility. But it is not inserting it as a timestamp thoough 
![image](https://github.com/strapi/strapi/assets/20578351/ce52251b-b035-4d32-aed0-d7d52c108ad2)

Reply:
arf ok let's test this on multiple DBs when we can

Reply:
I am just bothered there is not a way to insert a date with the other format! . As the plan is to also copy relations, I will not modify the `updated_at` value for now, just in case.